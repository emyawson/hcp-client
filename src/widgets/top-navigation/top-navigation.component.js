import React from 'react';

import { ProfileDropdown } from './profile-dropdown';
import { TopNavigationContainerDiv } from './top-navigation.style';

export const TopNavigation = ({ fullName, onSignOut }) => (
  <TopNavigationContainerDiv className="print-hide">
    <ProfileDropdown fullName={fullName} onSignOut={onSignOut} />
  </TopNavigationContainerDiv>
);
