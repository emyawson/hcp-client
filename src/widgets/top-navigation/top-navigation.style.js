import styled from 'styled-components';

import { spacing } from 'src/core';

export const TopNavigationContainerDiv = styled.div`
  padding-top: ${spacing.four};
  width: 100%;
  display: flex;
  min-height: 100px;
`;
TopNavigationContainerDiv.displayName = 'TopNavigationContainerDiv';
