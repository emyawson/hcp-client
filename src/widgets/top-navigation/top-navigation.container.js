import { compose } from 'recompose';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { mapDispatchers } from 'src/utils';
import { onSignOut } from 'src/core/authentication/authentication.actions';
import { selectFullName } from 'src/core/user/user.selectors';

import { TopNavigation } from './top-navigation.component';

export const topNavigationConnector = createStructuredSelector({
  fullName: selectFullName,
});

const dispatchers = mapDispatchers({
  onSignOut,
});

export const TopNavigationContainer = compose(
  connect(
    topNavigationConnector,
    dispatchers,
  ),
)(TopNavigation);
