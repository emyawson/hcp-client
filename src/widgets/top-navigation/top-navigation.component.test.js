import { shallow } from 'enzyme';
import React from 'react';

import { TopNavigation } from './top-navigation.component';

describe('top-navigation test suite', () => {
  test('it renders correctly', () => {
    const session = {
      user: {
        fullname: 'Jimmy Marlon Professional',
      },
    };
    const wrapper = shallow(<TopNavigation session={session} />);
    expect(wrapper).toMatchSnapshot();
  });
});
