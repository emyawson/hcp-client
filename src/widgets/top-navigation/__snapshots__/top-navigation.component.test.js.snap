// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`top-navigation test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    session={
      Object {
        "user": Object {
          "fullname": "Jimmy Marlon Professional",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <ProfileDropdown
        fullName={undefined}
        onSignOut={undefined}
      />,
      "className": "print-hide",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "fullName": undefined,
        "onSignOut": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <ProfileDropdown
          fullName={undefined}
          onSignOut={undefined}
        />,
        "className": "print-hide",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "fullName": undefined,
          "onSignOut": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
